random.go

```diff
@@ -0,0 +1,24 @@
+package chd
+
+import (
+	rand "github.com/remerge/go-xorshift"
+)
+
+func pickRandom(slice [][]byte) (v []byte) {
+	length := len(slice)
+	from := rand.Intn(length)
+	v = slice[from]
+	if len(v) > 0 {
+		return v
+	}
+
+	for l, r := from-1, from+1; l >= 0 || r < len(slice); l, r = l-1, r+1 {
+		if r < len(slice) && len(slice[r]) > 0 {
```

Comment:
Unlikely but what happens if `r` or `l` overflow?

Reply:
@akaspin an additional check won't hurt

Reply:
`from` can not be less or greater than length
`l` can not be greater than `from-1`
`r` can not be less than `from+1`

I'll make commit with tests and small improvements.